skore/src/skore/sklearn/_comparison/roc_curve_display.py

```diff
@@ -0,0 +1,304 @@
+    (0, (1, 5)),
+    (0, (1, 1)),
+    (5, (10, 3)),
+    (0, (5, 10)),
+    (0, (5, 5)),
+    (0, (5, 1)),
+    (0, (3, 10, 1, 10)),
+    (0, (3, 5, 1, 5)),
+    (0, (3, 1, 1, 1)),
+    (0, (3, 5, 1, 5, 1, 5)),
+    (0, (3, 10, 1, 10, 1, 10)),
+    (0, (3, 1, 1, 1, 1, 1)),
+]
+
+
+class RocCurveDisplay(HelpDisplayMixin, _ClassifierCurveDisplayMixin):
+    """ROC Curve visualization for comparison report.
+
+    An instance of this class is should created by `ComparisonReport.metrics.roc()`.
+    You should not create an instance of this class directly.
+
+    Parameters
+    ----------
+    fpr : dict of list of ndarray
+        False positive rate. The structure is:
+
+        - for binary classification:
+            - the key is the positive label.
+            - the value is a list of `ndarray`, each `ndarray` being the false
+              positive rate.
+        - for multiclass classification:
+            - the key is the class of interest in an OvR fashion.
+            - the value is a list of `ndarray`, each `ndarray` being the false
+              positive rate.
+
+    tpr : dict of list of ndarray
+        True positive rate. The structure is:
+
+        - for binary classification:
+            - the key is the positive label
+            - the value is a list of `ndarray`, each `ndarray` being the true
+              positive rate.
+        - for multiclass classification:
+            - the key is the class of interest in an OvR fashion.
+            - the value is a list of `ndarray`, each `ndarray` being the true
+              positive rate.
+
+    roc_auc : dict of list of float
+        Area under the ROC curve. The structure is:
+
+        - for binary classification:
+            - the key is the positive label
+            - the value is a list of `float`, each `float` being the area under
+              the ROC curve.
+        - for multiclass classification:
+            - the key is the class of interest in an OvR fashion.
+            - the value is a list of `float`, each `float` being the area under
+              the ROC curve.
+
+    estimator_names : str
+        Name of the estimators.
+
+    ml_task : str
+        Type of ML task.
+
+    pos_label : int, float, bool or str, default=None
+        The class considered as positive. Only meaningful for binary classification.
+
+    data_source : {"train", "test", "X_y"}, default=None
+        The data source used to compute the ROC curve.
+    """
+
+    def __init__(
+        self,
+        *,
+        fpr,
+        tpr,
+        roc_auc,
+        estimator_names,
+        ml_task,
+        pos_label=None,
+        data_source=None,
+    ):
+        self.fpr = fpr
+        self.tpr = tpr
+        self.roc_auc = roc_auc
+        self.estimator_names = estimator_names
+        self.ml_task = ml_task
+        self.pos_label = pos_label
+        self.data_source = data_source
+
+    def plot(
+        self,
+        *,
+        plot_chance_level=True,
```

Comment:
I would replicate the exact interface of the current roc curve display:

```python
        self,
        ax=None,
        *,
        estimator_name=None,
        roc_curve_kwargs=None,
        plot_chance_level=True,
        chance_level_kwargs=None,
        despine=True,
```

It is temporary fine to get a different display but we should make sure that a user do not see a different interface.